agent/src/main/java/reactor/blockhound/BlockHound.java

```diff
@@ -254,6 +258,58 @@ public TypePool typePool(ClassFileLocator classFileLocator, ClassLoader classLoa
 
         private Predicate<Thread> dynamicThreadPredicate = t -> false;
 
+        /**
+         * A phase of the builder where allowed blocking methods are configured for a given class.
+         * See {@link Builder#allowBlockingCallsInside(String)}.
+         * <p>
+         * Note that methods of this class directly mutate the {@link Builder} as if one was calling
+         * {@link Builder#allowBlockingCallsInside(String, String)}. One MUST specify at least one
+         * allowed method for it to have an effect on the configuration.
+         */
+        public class BuilderAllowSpec {
+
+            final String className;
+
+            private BuilderAllowSpec(String className) {
+                this.className = className;
+            }
+
+            /**
+             * Allow blocking calls inside the static initializer block of the currently configured
+             * class. Use {@link #and()} to exit the class-specific phase and return
+             * to the {@link Builder}.
+             *
+             * @return this {@link BuilderAllowSpec} for further configuration of the given class
+             * @see #and()
+             */
+            public BuilderAllowSpec forStaticInitializer() {
+                Builder.this.allowBlockingCallsInside(this.className, "<clinit>");
+                return this;
+            }
+
+            /**
+             * Allow blocking calls inside the specified method of the currently configured
+             * class. Use {@link #and()} to exit the class-specific phase and return
+             * to the {@link Builder}.
+             *
+             * @return this {@link BuilderAllowSpec} for further configuration of the given class
+             * @see #and()
+             */
+            public BuilderAllowSpec forMethod(String methodName) {
+                Builder.this.allowBlockingCallsInside(this.className, methodName);
+                return this;
+            }
+
+            /**
+             * Exit the class-specific phase and return to the {@link Builder}.
+             *
+             * @return the {@link Builder}
+             */
+            public Builder and() {
+                return Builder.this;
+            }
+        }
```

Comment:
What about constructors?

Reply:
ByteBuddy currently doesn't seem to support constructors, at least not with the way the agent is set up to apply allowances

Reply:
If there is nothing that can be done, this should at least be documented as a limitation to save the time if digging in, when the need arises (as it did in my case) only to find out it's not possible.

Reply:
would mentioning the limitation (and the #174 issue) in the root javadoc of `BuilderAllowSpec` work?